Subject: [PATCH] fix printk format specifiers and a const discard in warp

 - warp_main.c, warp_hw_v2.c: warp_dbg() calls passed heart_beat / ret with
   no matching specifier, so the values were silently dropped from the log.
 - warp_bm_v2.c: %ld printed a void * (desc->alloc_va).
 - warp_proc.c: len1 is size_t.
 - warp_wo.c: the masked __le32 field is wider than int here.
 - warp_fwdl.c: firmware->data is const in this kernel while bin_ptr stays
   writable for the loader.

diff -ru a/mcu/warp_fwdl.c b/mcu/warp_fwdl.c
--- a/mcu/warp_fwdl.c
+++ b/mcu/warp_fwdl.c
@@ -202,7 +202,7 @@
 		goto err_out;
 #endif	/* CONFIG_WARP_WO_EMBEDDED_LOAD */
 	} else {
-		ctrl->bin_ptr = ctrl->fw_entry->data;
+		ctrl->bin_ptr = (void *)ctrl->fw_entry->data;
 		ctrl->bin_size = ctrl->fw_entry->size;
 		ctrl->bin_mode = WO_FW_BIN;
 		ret = 0;
diff -ru a/mcu/warp_wo.c b/mcu/warp_wo.c
--- a/mcu/warp_wo.c
+++ b/mcu/warp_wo.c
@@ -466,7 +466,7 @@
 				rx_ctrl->dbg_ddone_check = 0;
 			}
 			} else {
-				warp_dbg(WARP_DBG_ERR, "wriong ddone bit 0x%x: %d ", cidx, (rxd->ctrl & WED_CTL_DMA_DONE));
+				warp_dbg(WARP_DBG_ERR, "wriong ddone bit 0x%x: %d ", cidx, (int)(rxd->ctrl & WED_CTL_DMA_DONE));
 			}
 		}
 
diff -ru a/regs/reg_v2/warp_hw_v2.c b/regs/reg_v2/warp_hw_v2.c
--- a/regs/reg_v2/warp_hw_v2.c
+++ b/regs/reg_v2/warp_hw_v2.c
@@ -3048,14 +3048,14 @@
 
 		if (ret < 0) {
 			warp_dbg(WARP_DBG_ERR, "%s(): wed_tx reset check fail,ret=%d\n", __func__, ret);
-			warp_dbg(WARP_DBG_ERR, "%s(): check ser_busy_cnt: echo 18 > wed\n", __func__, ret);
+			warp_dbg(WARP_DBG_ERR, "%s(): check ser_busy_cnt: echo 18 > wed, ret=%d\n", __func__, ret);
 		}
 		/* TX free done reset - rx path */
 		ret = reset_rx_traffic(wed, reset_type);
 
 		if (ret < 0) {
 			warp_dbg(WARP_DBG_ERR, "%s(): wed_rx reset check fail,ret=%d\n", __func__, ret);
-			warp_dbg(WARP_DBG_ERR, "%s(): check ser_busy_cnt: echo 18 > wed\n", __func__, ret);
+			warp_dbg(WARP_DBG_ERR, "%s(): check ser_busy_cnt: echo 18 > wed, ret=%d\n", __func__, ret);
 		}
 
 #ifdef WED_RX_D_SUPPORT
@@ -3064,7 +3064,7 @@
 
 		if (ret < 0) {
 			warp_dbg(WARP_DBG_ERR, "%s(): wed_rx_data check fail,ret=%d\n", __func__, ret);
-			warp_dbg(WARP_DBG_ERR, "%s(): check ser_busy_cnt: echo 18 > wed\n", __func__, ret);
+			warp_dbg(WARP_DBG_ERR, "%s(): check ser_busy_cnt: echo 18 > wed, ret=%d\n", __func__, ret);
 		}
 #endif
 
diff -ru a/warp_bm_v2.c b/warp_bm_v2.c
--- a/warp_bm_v2.c
+++ b/warp_bm_v2.c
@@ -1216,7 +1216,7 @@
 					desc->alloc_va,
 					desc->alloc_size, DMA_TO_DEVICE);
 	if (dma_mapping_error(&wed->pdev->dev, desc->alloc_pa)) {
-		warp_dbg(WARP_DBG_ERR, "%s(): FATAL, DMA map ERROR, %ld\n", __func__, desc->alloc_va);
+		warp_dbg(WARP_DBG_ERR, "%s(): FATAL, DMA map ERROR, %p\n", __func__, desc->alloc_va);
 		warp_os_free_mem(res->desc[idx].alloc_va);
 		ret = -1;
 		goto err;
diff -ru a/warp_main.c b/warp_main.c
--- a/warp_main.c
+++ b/warp_main.c
@@ -438,11 +438,11 @@
 		warp_fwdl_get_wo_heartbeat(&warp->woif.fwdl_ctrl, &heart_beat, warp->idx);
 		warp_dbg(WARP_DBG_ERR, "%s()wo heartbeat: 0x%x!\n", __func__, heart_beat);
 		/* dump WO PC LR log for 2 times*/
-		warp_dbg(WARP_DBG_ERR, "%s()======WO PC LR dump======\n", __func__, heart_beat);
+		warp_dbg(WARP_DBG_ERR, "%s()======WO PC LR dump====== hb:0x%x\n", __func__, heart_beat);
 		for (i = 0; i < 2; i++) {
 			warp_wo_pc_lr_cr_dump(warp->idx);
 		}
-		warp_dbg(WARP_DBG_ERR, "%s()======WO PC LR dump======\n", __func__, heart_beat);
+		warp_dbg(WARP_DBG_ERR, "%s()======WO PC LR dump====== hb:0x%x\n", __func__, heart_beat);
 	}
 #endif
 	warp_dbg(WARP_DBG_INF, "%s(): platform device probe is done\n", __func__);
diff -ru a/warp_proc.c b/warp_proc.c
--- a/warp_proc.c
+++ b/warp_proc.c
@@ -453,7 +453,7 @@
 	struct warp_entry *warp = (struct warp_entry *)PDE_DATA(file_inode(file));
 
 	if (len1 > 64 || len1 == 0) {
-		warp_dbg(WARP_DBG_OFF, "%s(): invalid length len1 %d!\n", __func__, len1);
+		warp_dbg(WARP_DBG_OFF, "%s(): invalid length len1 %zu!\n", __func__, len1);
 		goto end;
 	}
 
